source/index-management/index-management.rst

```diff
@@ -852,6 +852,181 @@ Example::
 	> a
 	[ "_id_", "ty_1", "l_2dsphere", "ts_1" ]
 
+--------------
+Search Indexes
+--------------
+
+Server 7.0 introduced three new server commands and a new aggregation stage to facilitate management of search indexes.  Drivers MUST provide 
+an API similar to the existing index management API specifically for search indexes.
+
+Search Index Management Helper Options
+--------------------------------------
+
+There are currently no supported options for any of the search index management commands.  To future proof
+drivers implementations so that any options added in the future do not constitute a breaking change to drivers,
+empty options structs have been added as placeholders.  If a driver's language has a mechanism to add options 
+in a non-breaking manner (i.e., method overloading) drivers MAY omit the empty options structs from their
+search index management helpers.
+
+``listSearchIndexes`` is implemented using an aggregation pipeline.  The list helper MUST support a driver's aggregation
+options.  Drivers MAY combine the aggregation options with
+any future ``listSearchIndexes`` stage options, if that is idiomatic for a driver's language.
+
+Notes
+-----
+
+The search index commands are asynchronous and return from the server before the index is successfully updated / created.
+In order to determine when an index has been created / updated, users are expected to run the ``listSearchIndexes`` repeatedly
+until index changes appear.
+
+An example, from Javascript:
+
+.. code:: typescript
+
+  const name = await collection.createSearchIndex({ definition: { ... fill out definition } })
+  while (!(await collection.listSearchIndexes({ name }).hasNext())) {
+    await setTimeout(1000);
+  }
+ 
+Common Interfaces
+-----------------
+
+.. code:: typescript
+
+  interface SearchIndexModel {
+    // Returns the definition for this index.
+    definition: Document;
+
+    // Returns the name for this index, if present.
+    name: Optional<string>;
+  }
+
+  /**
+   * The following interfaces are empty but are provided as placeholders for drivers that cannot 
+   * add options in a non-breaking manner, if options are added in the future.
+   */
+  interface CreateSearchIndexOptions {} 
+  interface UpdateSearchIndexOptions {}
+  interface ListSearchIndexOptions {}
+  interface DropSearchIndexOptions {}
+
+Standard API for Search Indexes
+-------------------------------
+
+.. code:: typescript
+
+  interface Collection {
+    /**
+    * Convenience method for creating a single search index.
+    * 
+    * @return The name of the created search index
+    * @note Drivers MAY opt to implement this method signature, the signature that
+    *   takes an SearchIndexModel as a parameter, or for those languages with method
+    *   overloading MAY decide to implement both.
+    */
+    createSearchIndex(name: String, definition: Document, options: Optional<CreateSearchIndexOptions>): String;
```

Comment:
Let's make this one optional and the other that takes a model required. Otherwise the unified tests, which use the method that takes a model would not align with drivers choosing to implement this one.

Reply:
Does it matter?  I assumed that each driver would transform the unified test operation into the correct format for their driver.  In Node, we could accommodate either:

```typescript

// index model API
operations.set('createSearchIndex', async ({ entities, operation }) => {
  const collection: Collection<any> = entities.getEntity('collection', operation.object);
  const { definition } = operation.arguments ?? {};
  return collection.createSearchIndex(definition);
});

// name + definition api
operations.set('createSearchIndex', async ({ entities, operation }) => {
  const collection: Collection<any> = entities.getEntity('collection', operation.object);
  const { definition: arguments } = operation.arguments ?? {};
  const { definition, name } = arguments ?? {};
  return collection.createSearchIndex(name, definition);
});
```

If we do end up supporting both, then maybe we should adjust the unified tests to be a bit more clearly accommodating of either approach, something like

```yaml
- operation: createSearchIndex
  arguments: 
    - indexName: <string>
    - indexDefinition: <string>
```

Reply:
I think that's fine with having the unified test suite modify the parameters for their driver. But I'd say we call that out explicitly somewhere.